segment.go

```diff
@@ -168,3 +168,19 @@ type DocVisitState interface {
 type StatsReporter interface {
 	ReportBytesWritten(bytesWritten uint64)
 }
+
+type FieldStatsReporter interface {
+	UpdateFieldStats(FieldStats)
+}
+
+type FieldStats interface {
+	Store(statName, fieldName string, value uint64)
+	Aggregate(stats FieldStats)
+	Fetch() map[string]map[string]uint64
+}
+
+const NumVecsStat = "num_vectors"
+
+var Stats = map[string]struct{}{
```

Comment:
No, let's not do this here. Drop the const from above as well. These are bleve/zap details.

Also the naming is way too generic.

Reply:
Yup, dropping these changes